packages/expo-modules-core/ios/AppDelegates/ExpoAppDelegate.swift

```diff
@@ -115,21 +115,28 @@ open class ExpoAppDelegate: UIResponder, UIApplicationDelegate {
     let selector = #selector(application(_:didReceiveRemoteNotification:fetchCompletionHandler:))
     let subs = subscribers.filter { $0.responds(to: selector) }
     var subscribersLeft = subs.count
-    var fetchResult: UIBackgroundFetchResult = .noData
     let dispatchQueue = DispatchQueue(label: "expo.application.remoteNotification", qos: .userInteractive)
+    var failedCount = 0
+    var newDataCount = 0
 
     let handler = { (result: UIBackgroundFetchResult) in
       dispatchQueue.sync {
         if result == .failed {
-          fetchResult = .failed
-        } else if fetchResult != .failed && result == .newData {
-          fetchResult = .newData
+          failedCount += 1
+        } else if result == .newData {
+          newDataCount += 1
```

Comment:
I would just change existing conditions to give `newData` a higher priority, but that's ok too 😉 

```swift
if result == .newData {
  fetchResult = .newData
} else if fetchResult != .newData && result == .failed {
  fetchResult = .failed
}
```

Reply:
this would incorrectly result in `.failed` callbacks if `result` would be `.noData`